node/network/gossip-support/src/lib.rs

```diff
@@ -150,23 +163,30 @@ impl State {
 		for leaf in leaves {
 			let current_index = util::request_session_index_for_child(leaf, ctx.sender()).await.await??;
 			let maybe_new_session = match self.last_session_index {
-				Some(i) if i >= current_index => None,
+				Some(i) if current_index <= i && !self.force_request => None,
 				_ => Some((current_index, leaf)),
 			};
 
 			if let Some((new_session, relay_parent)) = maybe_new_session {
 				tracing::debug!(target: LOG_TARGET, %new_session, "New session detected");
 				let authorities = determine_relevant_authorities(ctx, relay_parent).await?;
 				ensure_i_am_an_authority(keystore, &authorities).await?;
-				tracing::debug!(target: LOG_TARGET, num = ?authorities.len(), "Issuing a connection request");
+				let num = authorities.len();
+				tracing::debug!(target: LOG_TARGET, %num, "Issuing a connection request");
 
-				connect_to_authorities(
+				let failures = connect_to_authorities(
 					ctx,
 					authorities,
 					PeerSet::Validation,
 				).await;
 
+				// we await for the request to be processed
+				// this if fine, it should take much less time than one session
+				let failures = failures.await.unwrap_or(0);
+
 				self.last_session_index = Some(new_session);
+				// issue another request if at least half of the authorities were not resolved
+				self.force_request = failures != 0 && (num / failures <= 2);
```

Comment:
Not sure half is a good threshold. Reasoning:

- We assume approx. 2/3 honest validators, which is pretty vague but usually also means that we rely on 2/3 to be operational and reachable (GRANDPA).
- Even if lots of validators are offline, we would not really do any harm trying to connect to them every 6 seconds.

In summary, I would retry when we have less than 2/3+1 connectivity, even a higher threshold should be fine, as it would be reached normally and if not, querying a cache every 6 seconds should hardly cause problems.

Reply:
As a reference, we need backing groups to collectively directly connect to 2f+1 validators, but 1/2 each does not ensure anything.  We've no topology restrictions so they might both connect to the same 1/2.

Reply:
The comment says the half, but it's actually 1/3 + 1 failures, I'll change it to float for clarity. Also note that this is a temporary address resolution failure on startup, the connection request will get reissued on the next session anyway.
Also this is not about offlineness, but rather who's published their keys on DHT and how fast we quieried DHT.